docker/generic/Dockerfile.base

```diff
@@ -69,8 +69,12 @@ RUN groupadd --gid $GROUP_ID $USERNAME && \
         chmod 0440 /etc/sudoers.d/$USERNAME
 
 # Startup scripts
+ARG ROS_DISTRO
```

Comment:
ROS_DISTRO is already defined at the top of the file before the FROM statement.

Reply:
I think you need to repeat ARG if using it FROM as the first one had a scope of the following FROM command only - https://docs.docker.com/engine/reference/builder/#understand-how-arg-and-from-interact

Reply:
Just a minor thing, the packages for apt-get install should be in alphabetical order. Otherwise, the patch is a +1 from me on the Dockerfile changes.